vmdk_plugin/main_windows_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/Microsoft/go-winio"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+	"time"
+)
+
+const (
+	mock           = "mock"
+	requestPattern = "POST %s HTTP/1.0\r\n\r\n"
+	httpOK         = "HTTP/1.0 200 OK\r\n"
+)
+
+func TestNewPluginServer(t *testing.T) {
+	server := NewPluginServer(mock, nil)
+	assert.NotNil(t, server, "NewPluginServer should return a server instance")
+}
+
+func initPluginServer() *NpipePluginServer {
+	server := NewPluginServer(mock, nil)
+	go server.Init()            // server.Init() is a blocking call
+	time.Sleep(2 * time.Second) // wait for goroutine to execute
```

Comment:
Internally `server.Init()` calls `http.Serve(..)` on windows, which is similar to the `handler.ServeUnix(..)` call on linux i.e. both of them block until they're terminated explicitly. The `time.Sleep(..)` is just to make sure that the server has started. Do you think we should handle those calls in a different manner?